File: app/src/main/java/com/xyoye/dandanplay/utils/net/GzipInterceptor.java
package com.xyoye.dandanplay.utils.net;

import com.xyoye.dandanplay.utils.GZIPUtils;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.MediaType;
import okhttp3.Request;
import okhttp3.Response;
import okhttp3.ResponseBody;

public class GzipInterceptor implements Interceptor {

        @Override
        public Response intercept(Chain chain) throws IOException {
            Request request = chain.request();
            Response response = chain.proceed(request);
            if (response.code() == 200 && response.body() != null) {
                MediaType mediaType = response.body().contentType();
                byte[] data = response.body().bytes();
                if (GZIPUtils.isGzip(response.headers())) {
                    data = GZIPUtils.uncompress(data);
                }
                return response.newBuilder()
                        .body(ResponseBody.create(mediaType, data))
                        .build();
            } else {
                return response;
            }

        }
    }

File: common_component/src/main/java/com/xyoye/common_component/network/helper/GzipInterceptor.kt
package com.xyoye.common_component.network.helper

import com.xyoye.common_component.utils.GZIPUtils
import okhttp3.Interceptor
import okhttp3.Response
import okhttp3.ResponseBody.Companion.toResponseBody

/**
 * Created by xyoye on 2020/8/20.
 */

class GzipInterceptor : Interceptor{
    override fun intercept(chain: Interceptor.Chain): Response {
        val response = chain.proceed(chain.request())
        val responseBody = response.body
        if (response.code == 200 && responseBody != null){
            var data : ByteArray? = null
            if (GZIPUtils.isGzip(response.headers)){
                data = GZIPUtils.uncompress(responseBody.bytes())
            }
            if (data != null){
                return response.newBuilder()
                    .body(data.toResponseBody(responseBody.contentType()))
                    .build()
            }
        }
        return response
    }
}